app/models/manageiq/providers/kubevirt/inventory/parser.rb

```diff
@@ -212,8 +214,49 @@ def process_template(object)
     template_object.vendor = ManageIQ::Providers::Kubevirt::Constants::VENDOR
 
     # Add the inventory object for the hardware:
+    process_hardware(template_object, object.spec.template.spec.domain, metadata)
+
+    # Add the inventory object for the OperatingSystem
+    process_os(template_object, metadata)
+  end
+
+  def process_hardware(template_object, domain, metadata)
     hw_object = hw_collection.find_or_build(template_object)
-    hw_object.memory_mb = ManageIQ::Providers::Kubevirt::MemoryCalculator.convert(domain.memory, 'Mi')
+    hw_object.memory_mb = ManageIQ::Providers::Kubevirt::MemoryCalculator.convert(domain.resources.requests.memory, 'Mi')
```

Comment:
memory can become a quantity, thus it can have a suffix like "1Mi", "1MB", "2Gi", …

Reply:
MemoryCalcularor is handling all the cases. If it isn't please open an issue.